### PR TITLE
support alloc id 

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut  @disksing

### CODE REVIEW

**Comment by ngaut** [comment]
PTAL @disksing

File: server/id_test.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	}
+
+	var last uint64
+	for i := uint64(0); i < 2*allocStep; i++ {
+		msgID := uint64(rand.Int63())
+		sendRequest(c, conn, msgID, req)
+		msgID, resp := recvResponse(c, conn)
+		c.Assert(msgID, Equals, msgID)
+		c.Assert(resp.AllocId, NotNil)
+		c.Assert(resp.AllocId.GetId(), Greater, last)
```
Comment (disksing) [inline server/id_test.go]: Should check MetaType is match here.

File: server/id.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
+package server
+
+import (
+	"sync"
+
+	"golang.org/x/net/context"
+
```
Comment (disksing) [inline server/id.go]: Why splitting imports?

Comment (siddontang) [inline server/id.go]: goimport does it automatically.

File: server/id.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+		}
+
+		var (
+			cmp clientv3.Cmp
+			end uint64 = 0
+		)
+
+		if value == nil {
+			// create the key
+			cmp = clientv3.Compare(clientv3.CreatedRevision(key), "=", 0)
```
Comment (disksing) [inline server/id.go]: Why checking created version? I suppose it is ok if the key is accidentally created then removed before.

Comment (siddontang) [inline server/id.go]: for this case, the crated version is still 0.

Comment (siddontang) [inline server/id.go]: btw, I lose leader check here, and will add it.

Comment (disksing) [inline server/id.go]: I see.

**Comment by disksing** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**